src/Symmetric.jl

```diff
@@ -0,0 +1,213 @@
+@doc doc"""
+    SymmetricMatrices{N,T} <: Manifold
+
+The [`Manifold`](@ref) $ Sym (N)$ consisting of the `T`-valued, symmetric matrices of size $ n\times n$, i.e. the set 
+
+$\operatorname{Sym}(n) = \bigl\{A \in \mathbb F^{n\times n} \big| A^{\mathrm{T}} = A \bigr\}$
+
+where $\mathbb F \in \{ \mathbb R, \mathbb C\}$.
+For `T`, there is a distinction between 
+* `T = Real`: real-valued matrix entries
+* `T = Complex`: complex-valued matrix entries
+
+Being slightly inefficient, the matrices are safed as $n\times n$ arrays despite the redundance of the lower triangle.
+
+
+# Constructor
+
+    SymmetricMatrices(n::Int,T::Type = Real)
+
+"""
+struct SymmetricMatrices{N,T} <: Manifold end
+SymmetricMatrices(N::Int,T::Type = Real) = SymmetricMatrices{N,T}()
+
+function representation_size(::SymmetricMatrices{N,T}) where {N,T}
+    return (N,N)
+end
+
+@doc doc"""
+    manifold_dimension(M::SymmetricMatrices{n,T}) where {n,T <: Real}
+
+Return the dimension of the [`SymmetricMatrices`](@ref) matrix `M` with real-valued entries, i.e. 
+````math
+    \frac{n(n+1)}{2}
+````
+"""
+manifold_dimension(M::SymmetricMatrices{N,Real}) where {N,T} = div(N*(N+1),2)
+
+@doc doc"""
+    manifold_dimension(M::SymmetricMatrices{n,T}) where {n,T <: Complex}
+
+Return the dimension of the [`SymmetricMatrices`](@ref) matrix `M` with complex-valued entries, i.e. 
+````math
+    n(n+1)
+````
+"""
+manifold_dimension(M::SymmetricMatrices{N,Complex}) where {N,T} = N*(N+1)
+
+@doc doc"""
+    project_point!(M::SymmetricMatrices,x)
+    
+Implements the [`project_point`](@ref project_point!(M::Manifold, x)) of an arbitrary matrix `x` onto the manifold of [`SymmetricMatrices`](@ref), i.e. 
+
+$x_{new} = \frac{1}{2} \left( x_{old} + x_{old}^{\mathrm{T}} \right).$
+
+The value of the variable `x` is overwritten.
+"""
+project_point!(M::SymmetricMatrices, x) = (x .= 1/2 * (x + transpose(x)))
+
+@doc doc"""
+    project_tangent!(M::SymmetricMatrices, w, x, v)
+    
+Implements the [`project_tangent!`](@ref project_tangent!(M::Manifold, w, x, v)) of the matrix `v` onto the tangent space of the manifold of [`SymmetricMatrices`](@ref),
+i.e. the [`SymmetricMatrices`](@ref) itself. The result is stored in `w`:
+
+$w = \frac{1}{2} \left( v + v^{\mathrm{T}} \right).$
+
+"""
+project_tangent!(M::SymmetricMatrices, w, x, v) = (w .= 1/2 * (v + transpose(v)))
+
+
+@doc doc"""
+    inner(M::SymmetricMatrices, x, w, v)
+
+compute the inner product of the two tangent vectors `w,v` from the tangent
+plane at `x` on the [`SymmetricMatrices`](@ref) `M` using the restriction of the
+metric from the embedding, i.e. $ (v,w)_x = v^\mathrm{T}w $.
```

Comment:
Can you document this default metric in the docstring of `SymmetricMatrices`? Also I think this equation needs an `\operatorname{tr}`.
@kellertuer, what do you think about interpreting `EuclideanMetric` for matrix manifolds as being this metric? It seems be the only sensible interpretation to me, as our definition of `Euclidean` includes matrices that use that metric and then here we could specify the default metric as euclidean.

Reply:
What do you mean by “this” metric? The metric from the Euclidean space? The `v^Tw`? yes we could do that. I would call that the metric from the embedding, though.

also `$ (v,w)_x = v^\mathrm{T}w $.` should read
````
\[
(v,w)_x = \operatorname{tr}(v^{\mathrm{T}}w)
\]
````
here (@manuelweisser)

Reply:
Actually, nevermind I take that back. I thought we had given `Rotations` and `Sphere` the `EuclideanMetric` default, interpreted as them inheriting the metric from the embedding, but I see now that we didn't, so it's fine not to specify a default metric here.

Reply:
Did I get that comment right, @kellertuer  ? I changed that to 
```suggestion
using the restriction of the
metric from the embedding, i.e. 
````math
(v,w)_x = \operatorname{tr}(v^{\mathrm{T}}w)
````
```

Reply:
Concerning the math, yes :)